File: advent_of_code/day2/inventory-management-system.js
// Part 1

const fs = require('fs')
const path = require('path')

function calculateChecksum(file) {
  let ids = fs.readFileSync(file).toString().split('\n')
  let total = [0, 0]

  for (let line of ids) {
    let letters = {}
    for (let c of line) {
      if (letters[c]) letters[c] += 1
      else letters[c] = 1
    }
    if (Object.values(letters).includes(2)) total[0] += 1
    if (Object.values(letters).includes(3)) total[1] += 1
  }

  let checksum = total[0] * total[1]
  return checksum
}

let checksum = calculateChecksum(path.join(__dirname, 'day2.txt'))
console.log(checksum)

// Part 2

File: advent_of_code/day2/inventory-management-system.py
import os
from collections import Counter

# Part 1

def calculate_checksum(file):
    total = [0, 0]

    with open(file, 'r') as f:
        for line in f:
            line = line.strip()
            letter_counts = Counter(line).values()
            if 2 in letter_counts:
                total[0] += 1
            if 3 in letter_counts:
                total[1] += 1

    checksum = total[0] * total[1]
    return checksum

dir_path = os.path.dirname(os.path.abspath(__file__))
checksum = calculate_checksum(os.path.join(dir_path, 'day2.txt'))
print(checksum)

# Part 2

